src/components/gui/tabs/relational-diagram-tab.tsx

```diff
@@ -0,0 +1,214 @@
+import { DatabaseSchemaNode } from "@/components/database-schema-node";
+import { useSchema } from "@/context/schema-provider";
+import { DatabaseSchemas } from "@/drivers/base-driver";
+import { addEdge, Background, Connection, Controls, Edge, MiniMap, Node, ReactFlow, ReactFlowProvider, useEdgesState, useNodesState, useReactFlow } from '@xyflow/react';
+import '@xyflow/react/dist/style.css';
+import { useCallback, useEffect, useState } from "react";
+import { Toolbar } from "../toolbar";
+import { Button } from "@/components/ui/button";
+import { LucideRefreshCcw } from "lucide-react";
+import { Separator } from "@/components/ui/separator";
+import SchemaNameSelect from "../schema-editor/schema-name-select";
+import Dagre from '@dagrejs/dagre';
+import { AlignCenterHorizontalSimple, AlignCenterVerticalSimple } from "@phosphor-icons/react";
+import { DownloadImageDiagram } from "../export/download-image-diagram";
+
+function getLayoutElements(nodes: Node[], edges: Edge[], options: any) {
+  const g = new Dagre.graphlib.Graph().setDefaultEdgeLabel(() => ({}));
+  g.setGraph({ rankdir: options.direction });
+
+  edges.forEach((edge) => g.setEdge(edge.source, edge.target));
+  nodes.forEach((node) =>
+    g.setNode(node.id, {
+      ...node,
+      width: node.measured?.width ?? 0,
+      height: node.measured?.height ?? 0,
+    }),
+  );
+
+  Dagre.layout(g);
+
+  return {
+    nodes: nodes.map((node) => {
+      const position = g.node(node.id);
+      // We are shifting the dagre node position (anchor=center center) to the top left
+      // so it matches the React Flow node anchor point (top left).
+      const x = position.x - (node.measured?.width ?? 0) / 2;
+      const y = position.y - (node.measured?.height ?? 0) / 2;
+
+      // return { ...node }
+      return { ...node, position: { x, y } };
+    }),
+    edges,
+  };
+}
+
+function mapSchema(schema: DatabaseSchemas, selectedSchema: string): { initialNodes: Node[]; initialEdges: Edge[] } {
+  const initialNodes: unknown[] = [];
+  const initialEdges: Edge[] = [];
+
+  for (const item of schema[selectedSchema]) {
+    const items: unknown[] = [];
+    const relationShip = schema[selectedSchema].filter(x => x.tableSchema?.columns.filter(c => c.constraint?.foreignKey).map(c => c.constraint?.foreignKey?.foreignTableName).includes(item.name));
```

Comment:
I think it should be `relationship` not `relationShip`.

One more thing, foreign key can happen in `tableSchema.constraints`. There are two type of constraints for foreign key:

```
CREATE TABLE "main"."testing"(
  "id" INTEGER PRIMARY KEY,
  "agent_id" TEXT REFERENCES "agents"("agent_id") // <-- column constraint
)
```

We should also cover this case as well

```
CREATE TABLE "main"."testing"(
  "id" INTEGER PRIMARY KEY,
  "agent_id" TEXT,
  FOREIGN KEY ("agent_id") REFERENCES "agents" ("agent_id") // <-- this is table constraint
)
```